evalml/automl/automl_search.py

```diff
@@ -529,61 +547,6 @@ def __init__(
         self.data_splitter = data_splitter
         self.optimize_thresholds = optimize_thresholds
         self.ensembling = ensembling
-        if objective == "auto":
```

Comment:
This section was not deleted, simply moved down lower to be part of the other objective handling. It needed to be lower rather than moving the new handling up here because the new recommendation_objective logic requires several other validation checks to have already happened and `self.X_train` and `self.y_train` to be set.